### DISCUSSION TITLE
Compile Error: call to 'HTTPClient::begin' declared with attribute error: obsolete API

### DISCUSSION THREAD

**Discussion by focher** [Resinchem/Tailwind2MQTT#6] [Q&A] [answered]
When attempting to compile, I get the following error.

Version: Alpha Release v0.45

```
Arduino: 1.8.15 (Mac OS X), Board: "LOLIN(WEMOS) D1 mini (clone), 80 MHz, Flash, Disabled (new aborts on oom), Disabled, All SSL ciphers (most compatible), 32KB cache + 32KB IRAM (balanced), Use pgm_read macros for IRAM/PROGMEM, DOUT (compatible), 40MHz, 4MB (FS:2MB OTA:~1019KB), v2 Lower Memory, Disabled, None, Only Sketch, 921600"

/Tailwind2MQTT-main/tailwind_mqtt/tailwind_mqtt.ino: In function 'int getDoorStatus()':
tailwind_mqtt:437:13: error: call to 'HTTPClient::begin' declared with attribute error: obsolete API, use ::begin(WiFiClient, url)
  437 |   http.begin(doorStatusURL);
      |   ~~~~~~~~~~^~~~~~~~~~~~~~~
/Tailwind2MQTT-main/tailwind_mqtt/tailwind_mqtt.ino: In function 'void callback(char*, byte*, unsigned int)':
tailwind_mqtt:228:15: error: call to 'HTTPClient::begin' declared with attribute error: obsolete API, use ::begin(WiFiClient, url)
  228 |     http.begin(doorCommandURL);
      |     ~~~~~~~~~~^~~~~~~~~~~~~~~~
exit status 1
call to 'HTTPClient::begin' declared with attribute error: obsolete API, use ::begin(WiFiClient, url)
```

**Comment by Resinchem** [comment]
At first glance (with no real research at all!), I notice that the board your have selected is the 'LOLIN(WEMOS) D1 mini (clone)'.  A quick Google search shows some depreciated support for this board.  What actual board are you using?  If it is a D1 Mini can you try selecting WeMos D1 R1 and see if that compiles?  Let me know.  Another option is that you are using a newer version of one of the libraries than I am.  But let's eliminate the board issue first... and then look at libraries.

And thanks for testing and providing feedback!

**Comment by Resinchem** [comment]
OK.... after a bit more research, I believe the issue is related to a major version release (with a LOT of breaking changes) in the board manager for ESP8266.  The latest version is 3.0.0.  My build was based on 2.7.4... the latest before the major version change.  If possible, could you attempt to drop back your board manager for ESP8266 to v2.7.4 and see if that solves the compile problem?  I'm guessing it probably will.  On my end, I'll attempt to upgrade to 3.0.0... but with all the breaking changes, it may take me a while to resolve.

**Comment by Resinchem** [comment]
** **FIX RELEASED** **
v0.46 released with fix for the above http.begin obsolete error.  Just snag the new ino file from the release assets and replace your existing version.  Should (hopefully) compile with no errors.

**Comment by focher** [comment, marked as answer]
Here's how I ultimately used your info to get me there. I actually removed the ESP8266 board package (both 2.7.4 and 3.0.0) then followed the "Instructions - Other OS" steps to pull the latest from GitHub. I'm using a Mac (11.4 on M1 MacBook), so also was getting a Python error until I installed Python 3 as instructed on that page. Of course, I also pulled the 0.46 version down too.

From there, everything compiled and installed swimmingly. Unfortunately I'm traveling and won't be home for another week, but I do see the board advertising the "tailwind_ap" wireless network so I'm predicting I'm all good.

Just a few other points:

1. Using the "LOLIN(WeMos) D1 R1" board from the list.
2. The board I'm using is from [Amazon](https://www.amazon.com/gp/product/B081PX9YFV/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&psc=1).

**Reply by Resinchem** [reply]
Thanks for the feedback and I'm glad you got it compiled and uploaded.  I don't have a Mac to test on, so I'm glad you were able to figure out what it was... it's been bugging me as well as I couldn't find anything other than the small fix I put in 0.46.  Maybe I'll need to update the install info in the wiki for Macs?  But I'll sit tight and wait for you to return and do some more testing on your end.  Safe travels!

**Reply by focher** [reply]
Am back home now and re-did the process. This time, I just used the packages directly from the Arduino library manager - including v3.0.0. of the esp8266 package. With your code updates, everything worked perfectly.

**Comment by focher** [comment]
This was resolved with the 0.46 update.

**Reply by Resinchem** [reply]
Outstanding!  Thanks again for testing/feedback.  Are you saying that you successfully got the integration into Home Assistant (MQTT discovery) working as well, or that it just compiled?  I know that you were still waiting on the firmware update, so you probably haven't been able to confirm the actual Tailwind operation, but if I know that everything else is working properly, I can start thinking about next steps.  I didn't want to continue to make changes until we got your initial install working as expected.
